packages/back-end/src/routers/demo-datasource-project/new-demo-datasource-project.controller.ts

```diff
@@ -222,9 +223,13 @@ export const postDemoDatasourceProject = async (
   }
 
   try {
+    const projectName = "Sample Data";
+    const basePublicId =
+      generateProjectPublicIdFromName(projectName) || "sample-data";
     const project = await context.models.projects.create({
```

Comment:
We do this so many different places.  Should we just add this logic to the `beforeCreate` hook in the project model?  If `publicId` is empty, generate one from the name.